core/src/main/java/com/yahoo/ycsb/Utils.java

```diff
@@ -219,8 +219,25 @@ public static long getGCTotalCollectionCount() {
             ManagementFactory.getGarbageCollectorMXBeans();
         long count = 0;
         for (final GarbageCollectorMXBean bean : gcBeans) {
+          if (bean.getCollectionCount() < 0) {
+            continue;
+          }
           count += bean.getCollectionCount();
         }
         return count;
       }
+      
+      /** @return The total time, in milliseconds, spent in GC. */ 
+      public static long getGCTotalTime() {
+        final List<GarbageCollectorMXBean> gcBeans = 
+            ManagementFactory.getGarbageCollectorMXBeans();
+        long time = 0;
+        for (final GarbageCollectorMXBean bean : gcBeans) {
+          if (bean.getCollectionTime() < 0) {
+            continue;
+          }
+          time += bean.getCollectionTime();
```

Comment:
I was thinking of that too and even going into the MXbeans with a watcher to get the full cause, memory used/freed etc. I could try passing in the Measurement objects and massaging the collector names into spaceless keys. Should I go that route?